### DISCUSSION TITLE
Editor selection

### DISCUSSION THREAD

**Discussion by haztakki** [spatie/ray#68] [Ideas]
It would be nice if the **Editor** dropdown only contained installed applications rather than a lsit of commonly used ones. In addition to this it would be useful to be able to browse to a location, in case for whatever it cannot be found automatically.

**Comment by haztakki** [comment]
This idea expands on #59.

**Reply by AdrianMrn** [reply]
>It would be nice if the Editor dropdown only contained installed applications rather than a lsit of commonly used ones.

I'm not sure I like the privacy implications of this, I can imagine some users would dislike the idea of the app scanning their system for installed apps.

>In addition to this it would be useful to be able to browse to a location, in case for whatever it cannot be found automatically.

The problem with this approach is that it's not as easy as just starting an application, we have to use deeplinks specifically formatted by those apps. For example, PhpStorm and VSCode use slightly different syntax for this:

```
PhpStorm: 'phpstorm://open?file=%path&line=%line',
VSCode: 'vscode://file/%path:%line',
```

We could allow an option to add a custom deeplink here, but I think I'd rather just add on to the list until it's more complete. The list of actively-used editors is finite enough, I think.

Feel free to suggest some editors that are not in the list yet!

**Reply by haztakki** [reply]
Good points! I didn't think of those. Would it be possible to at least sort them by their installed status or give the user the option to hide them from the list? i.e. Untick Editor A, B and C.

**Reply by haztakki** [reply]
Even without an interface, a manual config file where a user can explicitly add and remove would be nice.

**Reply by AdrianMrn** [reply]
Good ideas. We'll discuss this internally when moving to the next big release!
